File: p730basicFileOperations/p730basicFileOperations/Person.m


#import "Person.h"


@implementation Person
@synthesize firstName, lastName;

- (void)dealloc {
    [firstName release];
    [lastName release];
    [super dealloc];
}

- (void)encodeWithCoder:(NSCoder *)encoder {
    //[super encodeWithCoder: encoder]; // not in this case
    [encoder encodeObject:self->lastName forKey:@"last"];
    [encoder encodeObject:self->firstName forKey:@"first"];
}

- (id) initWithCoder:(NSCoder *)decoder {
    //self = [super initWithCoder: decoder]; // not in this case
    self = [super init];
    self->lastName = [decoder decodeObjectForKey:@"last"];
    [self->lastName retain];
    self->firstName = [decoder decodeObjectForKey:@"first"];
    [self->firstName retain];
    return self;
}


@end

File: bk2ch23p818xml/ch36p1070xml/Person.swift


import UIKit

class Person: NSObject {
    // we're going to use KVC
    @objc var firstName : String
    @objc var lastName : String
    
    override var description : String {
        return self.firstName + " " + self.lastName
    }
    
    init(firstName:String, lastName:String) {
        self.firstName = firstName
        self.lastName = lastName
        super.init()
    }
    
}

